drivers/mtd_flashpage/mtd_flashpage.c

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2018 OTA keys S.A.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_mtd_flashpage
+ * @brief       Driver for internal flash devices implementing flashpage interface
+ *
+ * @{
+ *
+ * @file
+ * @brief       Implementation for the flashpage memory driver
+ *
+ * @author      Vincent Dupont <vincent@otakeys.com>
+ * @}
+ */
+
+#include <string.h>
+#include <errno.h>
+
+#include "mtd_flashpage.h"
+#include "periph/flashpage.h"
+
+static int _init(mtd_dev_t *dev)
+{
+    (void)dev;
+    assert(dev->pages_per_sector * dev->page_size == FLASHPAGE_SIZE);
+    return 0;
+}
+
+static int _read(mtd_dev_t *dev, void *buf, uint32_t addr, uint32_t size)
+{
+    (void)dev;
+    memcpy(buf, (void *)addr, size);
+
+    return size;
+}
+
+static int _write(mtd_dev_t *dev, const void *buf, uint32_t addr, uint32_t size)
+{
+    if (addr % FLASHPAGE_RAW_ALIGNMENT) {
+        return -EOVERFLOW;
```

Comment:
It doesn't look like an overflow, but rather like an invalid value.  Would returning -EINVAL make more sense here and in the check for `FLASHPAGE_RAW_ALIGNMENT` and `FLASHPAGE_RAW_BLOCKSIZE`.
The mtd API would need adaption though.

Reply:
addressed